pkg/backup/apimanager_backup.go

```diff
@@ -0,0 +1,346 @@
+	"SystemRecaptcha":     "system-recaptcha",
+	"SystemRedis":         "system-redis",
+	"SystemMemcached":     "system-memcache", // TODO should we backup/restore this one?
+	"Zync":                "zync",
+	"SystemMasterAPIcast": "system-master-apicast",
+}
+
+var configMapsToBackup map[string]string = map[string]string{
+	"SystemEnvironment":  "system-environment",
+	"APIcastEnvironment": "apicast-environment",
+}
+
+type APIManagerBackup struct {
+	options *APIManagerBackupOptions
+}
+
+func NewAPIManagerBackup(options *APIManagerBackupOptions) *APIManagerBackup {
+	return &APIManagerBackup{
+		options: options,
+	}
+}
+
+func (b *APIManagerBackup) Validate() error {
+	return nil
+}
+
+func (b *APIManagerBackup) APIManager() *appsv1alpha1.APIManager {
+	return b.options.APIManager
+}
+
+func (b *APIManagerBackup) BackupDestinationPVC() *v1.PersistentVolumeClaim {
+	if b.options.APIManagerBackupPVCOptions == nil {
+		return nil
+	}
+
+	volName := ""
+	if b.options.APIManagerBackupPVCOptions.BackupDestinationPVC.VolumeName != nil {
+		volName = *b.options.APIManagerBackupPVCOptions.BackupDestinationPVC.VolumeName
+	}
+
+	res := &v1.PersistentVolumeClaim{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "v1",
+			Kind:       "PersistentVolumeClaim",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      b.options.APIManagerBackupName,
+			Namespace: b.options.Namespace,
+		},
+		Spec: v1.PersistentVolumeClaimSpec{
+			AccessModes: []v1.PersistentVolumeAccessMode{
+				v1.ReadWriteOnce,
+			},
+			StorageClassName: b.options.APIManagerBackupPVCOptions.BackupDestinationPVC.StorageClass,
+			VolumeName:       volName, // TODO maybe make directly the pvc volume name option a string instead of a pointer to string? if we do that we would not be able to differentiate between not passed and passed but empty
+		},
+	}
+
+	if b.options.APIManagerBackupPVCOptions.BackupDestinationPVC.StorageRequests != nil {
+		res.Spec.Resources.Requests = v1.ResourceList{
+			v1.ResourceStorage: *b.options.APIManagerBackupPVCOptions.BackupDestinationPVC.StorageRequests,
+		}
+	}
+
+	return res
+}
+
+func (b *APIManagerBackup) BackupSecretsAndConfigMapsToPVCJob() *batchv1.Job {
+	if b.options.APIManagerBackupPVCOptions == nil {
+		return nil
+	}
+
+	var completions int32 = 1
+	return &batchv1.Job{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "batch/v1",
+			Kind:       "Job",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      fmt.Sprintf("job-backup-cfgmaps-secrets-%s", b.options.APIManagerBackupName),
+			Namespace: b.options.Namespace,
+		},
+		Spec: batchv1.JobSpec{
+			Completions: &completions,
+			// TODO BackoffLimit field controls how many times the job is retried
+			Template: v1.PodTemplateSpec{
+				Spec: v1.PodSpec{
+					Volumes: []v1.Volume{
+						b.pvcBackupDestinationPodVolume(),
+					},
+					Containers: []v1.Container{
+						v1.Container{
+							Name:  "backup-cfgmaps-secrets",
+							Image: "registry.redhat.io/openshift4/ose-cli:4.2",
```

Comment:
Yes they should. Because of the current state of the PR / solution I decided to code it that way at that moment. When we advance more in this PR let's discuss changing it before merging it :+1: